### PR TITLE
Refactor ImageNet wrapper such that it uses tensorflow/models

### LINKED ISSUE #425: Check Imagenet encoder for bugs
Dropout in vgg, batch normalization in resnet, etc.

### PR DESCRIPTION
**Pull request by jlibovicky**
The ImageNet wrapper is updated:

* the pre-trained networks are no longer from `tf.contrib.slim.nets`, but from [tensorflow/models repository](https://github.com/tensorflow/models) which needs to be cloned independently
* it is no longer possible to fine-tune the networks during training (there is no easy way, how to switch them between training/validation mode) and it in fact never worked
* `imagenet_reader` now has proper pre-processing for VGG nets and ResNet

Closes #425.

### CODE REVIEW

**Comment by jlibovicky** [comment]
Closes #425.

**Comment by jlibovicky** [comment]
Myslím, že je hotovo. @jindrahelcl?

File: neuralmonkey/encoders/imagenet_encoder.py [hunk 1]
```diff
@@ -131,22 +173,20 @@ def __init__(self,
     @tensor
     def input_image(self) -> tf.Tensor:
         return tf.placeholder(
-            tf.float32, [None, self.HEIGHT, self.WIDTH, 3])
+            tf.float32, [None, self.height, self.width, 3])
 
     @tensor
     def spatial_states(self) -> Optional[tf.Tensor]:
-        if self.attention_layer is None:
+        if self.spacial_layer is None:
```
Comment (jindrahelcl) [inline neuralmonkey/encoders/imagenet_encoder.py]: tohle se má jmenovat spatial ne spacial
